### PR TITLE
Cleanup MethodInfo tests

### PR DESCRIPTION
**Pull request by hughbe**
/cc @stephentoub

### CODE REVIEW

**Comment by stephentoub** [comment]
@dotnet-bot Test Innerloop OSX Release Build and Test please ("The plugin hasn't been performed correctly:") (@mmitche, FYI)

**Comment by karelz** [comment]
@AtsushiKan @stephentoub can you please review it?

**Comment by AtsushiKan** [comment]
DId you add or change tests or is this just moving their source into one .cs file? Because I don't quite understand what we gain by doing that.

**Comment by hughbe** [comment]
Tests were cleaned up. The reason I combined them into a single file was that they share a lot of common test data, which is part of the reason why the diff deletion count is large

**Comment by AtsushiKan** [comment]
Ok, I’ll merge this but I do think in the future, we should avoid mass source file migrations like this, especially if they’re combined with cleanup. It makes it hard to follow the evolution of the code.

**Comment by hughbe** [comment]
You're right - this code was 6 months old on my backlog and I had very little appetite to redo it, sorry. 
I'll bear this in mind for the future
